packages/react-reconciler/src/ReactFiberWorkLoop.js

```diff
@@ -1946,6 +1949,11 @@ function commitRootImpl(root, renderPriorityLevel) {
       nextEffect.nextEffect = null;
       nextEffect = nextNextEffect;
     }
+    // Clear the nextEffect pointers of an alternates
```

Comment:
Do we need the loop above then? Can we unify them?

Reply:
Unfortunately, we still need to keep the loop above. They're looping through separate things. The loop I've added is essentially going through all unmounted fibers and clearing the `nextEffect` pointer. There might be some effect fibers that are not unmounted that still need their effects clearing, which is what the above loop does.